lib/rust/ensogl/core/src/display/symbol/gpu/registry.rs

```diff
@@ -15,15 +19,43 @@ use crate::display::symbol::WeakSymbol;
 use crate::system::gpu::data::uniform::Uniform;
 use crate::system::gpu::data::uniform::UniformScope;
 use crate::system::gpu::Context;
-
+use crate::system::web;
+use crate::system::web::traits::*;
 
 
 // =============
 // === Types ===
 // =============
 
-pub type SymbolDirty = dirty::SharedSet<SymbolId, Box<dyn Fn()>>;
+pub type Dirty = dirty::SharedSet<SymbolId>;
+
+
+// ===============
+// === RunMode ===
+// ===============
+
+/// The application run mode. It can be set to either [`Normal`] or [`ShaderExtraction`] mode. In
+/// the latter case some warnings are suppressed. For example, there will be no warning that
+/// precompiled shapes shaders were not found, as they are yet to be generated.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
+#[allow(missing_docs)]
+pub enum RunMode {
+    #[default]
+    Normal,
+    ShaderExtraction,
+}
 
+impl RunMode {
+    /// Check whether the mode is set to [`Normal`].
+    pub fn is_normal(self) -> bool {
+        self == Self::Normal
+    }
+
+    /// Check whether the mode is set to [`ShaderExtraction`].
+    pub fn is_shader_extraction(self) -> bool {
+        self == Self::ShaderExtraction
+    }
```

Comment:
Why? I don't want to import this module there and I just need to do one comparison. So I'm calling a method instead of having full import and more complex syntax there.

Reply:
is_none makes sense, because 
1. Option is widely and often used
2. not every Option implements PartialEq, and is_none is simpler than making pattern match.

In this case it's just another name I need to remember or ctrl+click to see what normal modes are.

Reply:
I still much prefer this code over adding additional imports. With imports you have the same issues - you need to ctrl+click to see what modes are, but the code is just more complex. Can we leave it this way?